pulpcore.fc

```diff
@@ -17,12 +17,7 @@
 /usr/local/lib/pulp/bin/gunicorn	--	gen_context(system_u:object_r:pulpcore_server_exec_t,s0)
 /usr/local/lib/pulp/bin/rq		--	gen_context(system_u:object_r:pulpcore_exec_t,s0)
 
-/var/lib/pulp/artifact(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
-/var/lib/pulp/assets(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
-/var/lib/pulp/devel(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
-/var/lib/pulp/tmp(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
-/var/lib/pulp/upload(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
-/var/lib/pulp/sign-metadata.sh	--	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
+/var/lib/pulp(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
```

Comment:
This will break installations that have both Pulp2 and Pulp3 installed :(

Reply:
@evgeni Thanks for bringing in this knowledge. I haven't took pulp2 in consideration.

Could you explain me a bit more what would break ? So I can see how to update this PR accordingly.
Would something like the following not break it ?

```
/var/lib/pulp 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
/var/lib/pulp/pulpcore_static(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
```

Reply:
See https://github.com/pulp/pulpcore-selinux/pull/15#discussion_r490136560 for the file contexts provided. The relevant one is:
```
/var/lib/pulp(/.*)?                                all files          system_u:object_r:httpd_sys_rw_content_t:s0 
```

Multiple SELinux policies can't specify the same regex.

The reason `/var/lib/pulp` is used (as opposed to `/var/lib/pulpcore`) is to allow a migration from Pulp 2. Many Satellite users have a dedicated `/var/lib/pulp` partition. Moving files across a partition boundary is very expensive so it was chosen to put all files in `/var/lib/pulp`.

Reply:
@ekohl to make sure I understand. In order to be compliant with both this patch that currently states:

```
/var/lib/pulp(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
```

should be updated to the following ?

```
# This was already there
/var/lib/pulp/artifact(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
/var/lib/pulp/assets(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
/var/lib/pulp/devel(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
/var/lib/pulp/tmp(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
/var/lib/pulp/upload(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
/var/lib/pulp/sign-metadata.sh	--	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)

# New fc
/var/lib/pulp(/.*)? 	gen_context(system_u:object_r:httpd_sys_rw_content_t,s0)
/var/lib/pulp/pulpcore_static(/.*)? 	gen_context(system_u:object_r:pulpcore_var_lib_t,s0)
```

Reply:
No, it can't specify `/var/lib/pulp(/.*)?` since that's present in `pulp-selinux` (the version 2). I haven't been able find the sources for that so I can't easily link them. The fact https://github.com/pulp/pulp-selinux redirects to this repo makes me suspect they have been destroyed but it's also possible it was never hosted in this org.

That means it should only add `pulpcore_static`. I do wonder why that new directory is introduced. Can you tell me where it comes from?

Reply:
The pulp2 policy you're searching for is here: https://github.com/pulp/pulp/blob/2-master/server/selinux/server/pulp-server.fc

Reply:
@ekohl I have no idea. I don't know pulp enough. Its there on a pulp3 / galaxy_ng server I deployed.

/me updates PR